### DISCUSSION TITLE
What is the largest number of interfaces implemented on a single object in `dotnet/runtime`?

### DISCUSSION THREAD

**Discussion by tacosontitan** [dotnet/runtime#92801] [Q&A] [answered]
This is purely a curiosity, but I was looking at the documentation for `BigInteger` and noticed just how many interfaces it's implementing:
![image](https://github.com/dotnet/runtime/assets/65432314/7f657a1c-40eb-4f5c-a788-295c57fbf558)

I *assume* that this shows the entire hierarchy since here in `dotnet/runtime` the implementation list isn't as verbose:

https://github.com/dotnet/runtime/blob/62d2c91b78a258e976db78e926e9e26b606214f1/src/libraries/System.Runtime.Numerics/src/System/Numerics/BigInteger.cs#L17-L24

This got me thinking however, does anyone have a best guess of the object with the most implementations as defined in `dotnet/runtime` (not full hierarchy)?

**Comment by DaZombieKiller** [comment, marked as answer]
[The answer appears to be `double`](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEUCuA7AHwAEAmARgFgAoIgBgAIiyA6AGQEs8BHAbmrsZkUfGgybMASvgzsAtjGYBhCLIAO7ADYwoAZW0A3dmBgBnEdX0BDKPQwBPVTHoBeegEFVqgCIrLnJThQsHgYPrJ+eMwA4jAYbiYmMLLAGuymABQAlNT0ufTMelpgGACylnh26ZYJSSl2LgB89NWJyRp20bEAKg4ZmdlUefkA6gAW2jDp9o6Ntr3MAHKW8iaqlsb07Cb0AN4AvvR42wBkx4cmBRjWGCbD7Bij6QBETE/9OXnMZQjpeDAA7vQAJIhbQAM3WMGU0mUams2iymQAhOYaGQAJzpAAkTx20wUADEcBoNEt5Hs0Lt8Z0MCCMODISYsmwYHgAOYPA6celQCHGC5vVGJSxaAAmjBIwNBvMh0JCsLWsBsIGBCvhUAAPD1HA1qDsPrlVFB2FZ6fRYJZRRA8O16Mo8KL7uxrSLhjBLABrLqWFIwLW9SnamAAbQAuk0APrchn8lyHAFZVFDIgAZk2ITtKkVkyDAH5moHevngAMhvrBkM8lYbOxLHGo9K+aYaQA1EU4SaWJGUmK0xuMzIiStV6ybYD16My/mt9uTYDd+i9ukx0yDg2VogAdk2lhZ7IeSiz8K6EHS7GAe45ozXFdye3X66NJssZqYSHoQbDi9iy6nGSDcyOKWeTlsOuRboBChLv2/KJuu95UAh1BAA===), at 37 interfaces. Notably though, it's not the _only_ type with this many interfaces (`float` is another) -- most of the primitives with support for generic math implement the same number of interfaces or something close to it.
